_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -369,6 +369,8 @@ public function get_module( $request ) {
 				&& ( new Status() )->is_offline_mode()
 			) {
 				$module['activated'] = false;
+			} elseif ( Jetpack::is_module_active( $request['slug'] ) ) { // TODO: Move into `Jetpack::get_module()`?
```

Comment:
This is an odd one, based on https://github.com/Automattic/jetpack/blob/e5a81d6e74516d2879f16caf9b0e2fac32bb16af/class.jetpack-admin.php#L136 I would have expected to have that information already?

Reply:
Yeah, I was surprised, too. (Forgot to leave a comment here, apologies.)

The line you found is used for the modules _list_ endpoint (`/modules/all`), but not for the individual one (`/modules/shortcodes`). (And I didn't want to use the list 'cause it's pretty big.)

**Here's the a call stack (of sorts) for the list endpoint:**

https://github.com/Automattic/jetpack/blob/1527e0be4d25312d5fb4672a3f856e11ad6d84db/_inc/lib/class.core-rest-api-endpoints.php#L276-L281
https://github.com/Automattic/jetpack/blob/1527e0be4d25312d5fb4672a3f856e11ad6d84db/_inc/lib/class.core-rest-api-endpoints.php#L74
https://github.com/Automattic/jetpack/blob/acfd44811fd55e3ead09648ab95adae92ce860b5/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php#L166
https://github.com/Automattic/jetpack/blob/acfd44811fd55e3ead09648ab95adae92ce860b5/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php#L177-L180
https://github.com/Automattic/jetpack/blob/acfd44811fd55e3ead09648ab95adae92ce860b5/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php#L192-L195
https://github.com/Automattic/jetpack/blob/acfd44811fd55e3ead09648ab95adae92ce860b5/class.jetpack-admin.php#L136

---

**Compare that to the individual module endpoint:**

https://github.com/Automattic/jetpack/blob/1527e0be4d25312d5fb4672a3f856e11ad6d84db/_inc/lib/class.core-rest-api-endpoints.php#L308-L312
https://github.com/Automattic/jetpack/blob/1527e0be4d25312d5fb4672a3f856e11ad6d84db/_inc/lib/class.core-rest-api-endpoints.php#L73
https://github.com/Automattic/jetpack/blob/1527e0be4d25312d5fb4672a3f856e11ad6d84db/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php#L322
https://github.com/Automattic/jetpack/blob/1527e0be4d25312d5fb4672a3f856e11ad6d84db/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php#L334-L338
https://github.com/Automattic/jetpack/blob/1527e0be4d25312d5fb4672a3f856e11ad6d84db/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php#L359-L362

`Jetpack::get_module()` doesn't add the `activated` field.

Reply:
As my `// TODO: Move into Jetpack::get_module()?` inline comment states, I was considering to move the `activated` field there. I'm not totally sure about the rationale behind decisions here; oversight aside, here are some hypotheses/considerations:

- `Jetpack::get_module()` should only expose 'static' information -- stuff that's encoded in the module's code, and doesn't change through user interaction.
- `Jetpack::get_module()` sits at a pretty low level. There might be unit or snapshot tests that could be easily thrown off by a newly added field.

Since I don't want this change to blow up the size of this PR (potentially needing to update tests, and research the full impact), I'm leaning towards keeping the change as-is, with the TODO as a reminder to potentially revisit.